src/bundle-manifest.ts

```diff
@@ -0,0 +1,160 @@
+/**
+ * A bundle is a grouping of files which serve the need of one or more
+ * entrypoint files.
+ */
+export interface IBundle {
+  entrypoints: Set<URL>;
+  files: Set<URL>;
+}
+
+/**
+ * A bundle strategy function is used to transform an array of bundles.
+ */
+export type IBundleStrategy = (bundles: IBundle[]) => IBundle[];
+
+/**
+ * A mapping of URLs to their full set of transitive dependencies, such that
+ * `a->b, b->c, b->d, e->f` would be represented `{a:[b,c,d], e:[f]}`
+ */
+export type IFileTransDepsIndex = Map<URL, Set<URL>>;
```

Comment:
I went with TransitiveDependenciesMap not to be redundant, but I think because it didn't feel like a _thing_ for some reason without it.  Definitely more readable than IFileTransDepsIndex, so good call.